packages/amplify-graphql-api-construct/API.md

```diff
@@ -0,0 +1,145 @@
+## API Report File for "@aws-amplify/graphql-api-construct-alpha"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { CfnApiKey } from 'aws-cdk-lib/aws-appsync';
+import { CfnDataSource } from 'aws-cdk-lib/aws-appsync';
+import { CfnFunctionConfiguration } from 'aws-cdk-lib/aws-appsync';
+import { CfnGraphQLApi } from 'aws-cdk-lib/aws-appsync';
+import { CfnGraphQLSchema } from 'aws-cdk-lib/aws-appsync';
+import { CfnPolicy } from 'aws-cdk-lib/aws-iam';
+import { CfnResolver } from 'aws-cdk-lib/aws-appsync';
+import { CfnResource } from 'aws-cdk-lib';
+import { CfnRole } from 'aws-cdk-lib/aws-iam';
+import { CfnTable } from 'aws-cdk-lib/aws-dynamodb';
+import { Construct } from 'constructs';
+import { Duration } from 'aws-cdk-lib';
+import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { IFunction } from 'aws-cdk-lib/aws-lambda';
+import { IRole } from 'aws-cdk-lib/aws-iam';
+import { IUserPool } from 'aws-cdk-lib/aws-cognito';
+import { ResolverConfig } from '@aws-amplify/graphql-transformer-core';
+import { SchemaFile } from 'aws-cdk-lib/aws-appsync';
+import { TransformerPluginProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import { TransformParameters } from '@aws-amplify/graphql-transformer-interfaces';
+
+// @public (undocumented)
+export class AmplifyGraphQlApi extends Construct {
+    constructor(scope: Construct, id: string, props: AmplifyGraphQlApiProps);
+    // (undocumented)
+    readonly env: string;
```

Comment:
This is, today we require the env to be passed in because it's used as a param in the cloudformation template, meaning we need it in the import. Until that's refactored I'm providing a mechanism for users to set that variable since it influences things like internal resource names, references, etc.

Reply:
Sounds good but we should hide property.
I.e. I shouldn't be able to `new GrapqhConstruct(...).env` .
If this property needs public read access for colaborating objects (transformers?) can this be obscured in some way?

Reply:
Ahh, sorry I think I misunderstood your earlier comment, I'll yank from the output object until we know we need it, or have a concrete use case. I was using this in my POCs to differentiate CfnOutput names.